external/CMakeLists.txt

```diff
@@ -113,5 +113,11 @@ add_library(secp256k1 STATIC IMPORTED GLOBAL)
 set_property(TARGET secp256k1
   PROPERTY IMPORTED_LOCATION ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libsecp256k1.a)
 
-add_library(CMSIS INTERFACE)
-target_include_directories(CMSIS INTERFACE CMSIS/Include)
+if(CMAKE_CROSSCOMPILING)
+  add_library(CMSIS INTERFACE)
```

Comment:
for my information, what does this do? why is there a library target with no c files?

Reply:
Those are the headers provided by ARM for cortex-m4 development. Full description is here: https://www.arm.com/why-arm/technologies/cmsis

The point of creating an interface-only library is just build-system cleanliness. Separate dependencies should show up as different entities in the graph. Makes it easier to reason about.